Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Use latest node LTS base image
+FROM node:lts
+
+# Clone the latest community server & install
+RUN git clone https://github.com/solid/community-server.git
+
+WORKDIR /community-server
+
+RUN npm ci
+
+RUN npm run build
+
+# Container config & data dir for volume sharing
+# Defaults to filestorage with /data directory (passed through CMD below)
+RUN mkdir /config && mkdir /data
+
+# Informs Docker that the container listens on the specified network port at runtime
+EXPOSE 3000
```

Comment:
That's true, but as per the documentation:

> The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published.

In other words, it's just a hint. Which materializes nicely for example when using the Docker desktop UI:

<img width="561" alt="Screenshot 2021-02-01 at 14 22 14" src="https://user-images.githubusercontent.com/613619/106470974-1204a400-6499-11eb-9170-801c06889da5.png">

... and doesn't stop anyone from running anything and everything from and to any port they want. I'd be curious to know why someone would run the container with another port than the defaults, if you have a use case in mind, could you share?

Reply:
I did have a use case in mind, but turns out I misunderstood what EXPOSE does. I almost never use it, and thought it actually exposed that port, rather than just signaling to users they probably want to expose that port. This is definitely the right way to do it from a documentation perspective.